### ISSUE TITLE
Feature: Bulk handling

### ISSUE THREAD

**Issue by MGatner** [codeigniter4/settings#8]
Performance-wise it would be a good idea to implement some bulk handling, so multiple settings can be retrieved with e.g. a single database call. For storage this might mean deferred writes, which shouldn't be too hard since we already use a local store.

**Comment by lonnieezell** [comment]
Agreed. I wasn't going to let this hold up a 1.0 release, though. We already limit the queries to a single read so keep performance strong there.
